code/obj/sawflymisc.dm

```diff
@@ -0,0 +1,641 @@
+	var/atk_delay = 5
+	var/attack_cooldown = 10
+	var/seekrange = 15
+	var/attacker = null
+	var/atom/movable/target = null
+	var/oldtarget_name = null
+	var/attack = 0
+	var/task = "thinking"
+	var/frustration = 0
+	var/atom/target_lastloc = null
+	var/aggressive = FALSE
+	var/last_found = null
+	var/steps = 0
+	var/sleep_check = 10
+	var/wander_check = 0
+	var/can_revive = TRUE
+	var/atksilicon = TRUE
+	var/atkcarbon = TRUE
+	var/alive = TRUE
+	var/alreadydead = FALSE // prevents any bullshit from happening with repeated
+
+
+	// MOB/LIVING/CRITTER FUCK SHIT HERE
+	setup_healths()
+		add_hh_robot(20, 1) // fuck you and your way of doing health that isn't my way of doing it
+		add_hh_robot_burn(20, 1)
+	custom_gib_handler = /proc/robogibs
+	blood_id = "oil"
+	use_stamina = FALSE
+	use_stunned_icon = FALSE
+	butcherable = FALSE
+	can_bleed = FALSE
+	canbegrabbed = FALSE
+	can_lie = FALSE
+	can_burn = FALSE
+	pet_text = "cuddles"
+
+
+	New()
+		..()
+		if(isnew)
+			name = "Sawfly [pick(sawflynames)]-[rand(1,999)]"
+		deathtimer = rand(1, 5)
+		death_text = "[src] jutters and falls from the air, whirring to a stop."
+		beeptext = "[pick(list("beeps",  "boops", "bwoops", "bips", "bwips", "bops", "chirps", "whirrs", "pings", "purrs", "thrums"))]"
+		animate_bumble(src) // gotta get the float goin' on
+		src.set_a_intent(INTENT_HARM) // incredibly stupid way of ensuring they aren't passable
+
+
+
+	proc/foldself()
+		var/obj/item/old_grenade/spawner/sawfly/reused/N = new /obj/item/old_grenade/spawner/sawfly/reused(get_turf(src))
+		// pass our name and health
+		N.name = "Compact [name]"
+		N.tempname = src.name
+		N.temphp = (src.get_health_percentage()) / 2
+		qdel(src)
+
+	proc/communalbeep() // distribues the beepchance among the number of sawflies nearby
+		fliesnearby = 1 //that's you, little man! :)
+		for(var/mob/living/critter/sawfly/E in range(get_turf(src), 18))
+			src.fliesnearby += 1 //that's your buddies!
+		var/beepchance = (1 / fliesnearby) * 100 //have it so that, if there's 2 sawflies, it's a 1/2 chance for each, then convert to %
+		if(fliesnearby<3) beepchance -=20
+		if(prob(beepchance))
+			playsound(src, pick(src.beeps), 40, 1)
+			src.visible_message("<b>[src] [beeptext].</b>")
+
+
+
+	emp_act() //same thing as if you emagged the controller, but much higher chance
+		if(prob(80))
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to sprial out of contro!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+	Cross(atom/movable/mover) //code that ensure projectiles hit them when they're alive, but won't when they're dead
+		if(istype(mover, /obj/projectile))
+			if(!src.alive)
+				return 1
+		return ..()
+
+
+	Uncross(atom/movable/O, do_bump = TRUE)
+		if(istype(O, /obj/projectile))
+			. = 1
+		return ..()
+
+
+	death() //FUCK YOU
+		CritterDeath() // FUCK YOU TOO
+
+	proc/CritterDeath() //  SUPER important proc do NOT touch this or everything will break and You Will Cry
```

Comment:
I think I'm going to stop reviewing here, because this code appears to be a mess of copy pasted/ported objcritter code that I honestly don't care to sort through. or merge in this state.